src/transformers/models/dpt/modeling_flax_dpt.py

```diff
@@ -0,0 +1,1236 @@
+
+        return hidden_state + residual
+
+
+class FlaxDPTFeatureFusionLayerCollection(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.layers = [
+            FlaxDPTFeatureFusionLayer(self.config, name=str(i)) for i in range(len(self.config.neck_hidden_sizes))
+        ]
+
+    def __call__(self, hidden_states, residual=None, i=0):
+        return self.layers[i](hidden_states, residual)
+
+
+class FlaxDPTFeatureFusionLayer(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32
+    align_corners: bool = True
+
+    def setup(self):
+        self.projection = nn.Conv(self.config.fusion_hidden_size, kernel_size=(1, 1))  # , bias=True)
+
+        self.residual_layer1 = FlaxDPTPreActResidualLayer(self.config)
+        self.residual_layer2 = FlaxDPTPreActResidualLayer(self.config)
+        self.upsample = FlaxDPTUpsample()
+
+    def __call__(self, hidden_state, residual=None):
+        if residual is not None:
+            if hidden_state.shape != residual.shape:
+                size = hidden_state.shape
+                residual = self.upsample(residual, size)
+            hidden_state = hidden_state + self.residual_layer1(residual)
+
+        hidden_state = self.residual_layer2(hidden_state)
+        hidden_state = self.upsample(hidden_state)
+        hidden_state = self.projection(hidden_state)
+
+        return hidden_state
+
+
+class FlaxDPTViTLayer(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32  # the dtype of the computation
+
+    def setup(self):
+        self.attention = FlaxDPTViTAttention(self.config, dtype=self.dtype)
+        self.intermediate = FlaxDPTViTIntermediate(self.config, dtype=self.dtype)
+        self.output = FlaxDPTViTOutput(self.config, dtype=self.dtype)
+        self.layernorm_before = nn.LayerNorm(epsilon=self.config.layer_norm_eps, dtype=self.dtype)
+        self.layernorm_after = nn.LayerNorm(epsilon=self.config.layer_norm_eps, dtype=self.dtype)
+
+    def __call__(self, hidden_states, deterministic: bool = True, output_attentions: bool = False):
+        attention_outputs = self.attention(
+            self.layernorm_before(hidden_states),  # in ViT, layernorm is applied before self-attention
+            deterministic=deterministic,
+            output_attentions=output_attentions,
+        )
+
+        attention_output = attention_outputs[0]
+
+        # first residual connection
+        attention_output = attention_output + hidden_states
+
+        # in ViT, layernorm is also applied after self-attention
+        layer_output = self.layernorm_after(attention_output)
+
+        hidden_states = self.intermediate(layer_output)
+        hidden_states = self.output(hidden_states, attention_output, deterministic=deterministic)
+
+        outputs = (hidden_states,)
+
+        if output_attentions:
+            outputs += (attention_outputs[1],)
+        return outputs
+
+
+class FlaxDPTViTLayerCollection(nn.Module):
```

Comment:
```suggestion
# Copied from transformers.models.vit.modeling_flax_vit.FlaxViTLayerCollection with ViTConfig->DPTConfig, ViTLayer->DPTViTLayer
class FlaxDPTViTLayerCollection(nn.Module):
```

Reply:
This does seem to not work in : 791ea245ed3470463d3a9e6fee2c49d384a4a71a will investigate that!

Reply:
Actually I think that we cannot copy it from `FlaxViTLayerCollection` since to do properly the forward pass you would need all the hidden states for each stage. On `FlaxViTLayerCollection` the tuple `all_hidden_states` is not returned which is what we use [here](https://github.com/younesbelkada/transformers/blob/f945eef63b733bd247b0d5d7c83a9fa28f06467c/src/transformers/models/dpt/modeling_flax_dpt.py#L1010) at the index 1